Libraries/Image/RCTImageLoader.m

```diff
@@ -382,11 +382,13 @@ - (RCTImageLoaderCancellationBlock)_loadImageOrDataWithURLRequest:(NSURLRequest
   });
 
   return ^{
-    if (cancelLoad && !cancelled) {
-      cancelLoad();
-      cancelLoad = nil;
-    }
-    OSAtomicOr32Barrier(1, &cancelled);
+    dispatch_async(_URLRequestQueue, ^{
```

Comment:
Looks reasonable but could you structure it like this? Also, you'll need to use weakSelf / strongSelf to access `_URLRequestQueue`? Right now this will retain self.

```
return ^{
  typeof(self) strongSelf = weakSelf;
  if (strongSelf && !cancelled) {
    dispatch_async(strongSelf->_URLRequestQueue, ^{
      if (cancelLoad) {
        cancelLoad();
        cancelLoad = nil;
     }
  });
OSAtomicOr32Barrier(1, &cancelled);
};